src/main/webapp/js/calendar.js

```diff
@@ -13,22 +13,187 @@
+// TODO: Add ability to remove courses from the calendar
+// TODO: Add ability to choose which section the user wants, not just
+// automatically
+//  choosing the first section
+
+export const Calendar = (() => {
+  let id = 1;  // The id of the next scheudle that will be added to the calendar
+  let cal;     // The calendar object
+  const enum_days = {
+    DATE_MONDAY: 0,
+    DATE_TUESDAY: 1,
+    DATE_WEDNESDAY: 2,
+    DATE_THURSDAY: 3,
+    DATE_FRIDAY: 4,
+  };
+
+  /**
+   * Initializes the calendar and moves the view to a hardcoded date in the
+   * past.
+   */
+  function initCalendar() {
+    cal = new tui.Calendar('#calendar', {
+      defaultView: 'week',
+      useCreationPopup: true,
+      useDetailPopup: true,
+      disableDblClick: true,
+      disableClick: true,
+      isReadOnly: true,
+      scheduleView: ['time'],
+      taskView: false,
+      week: {
+        workweek: true,
+      },
+      timezones: [{
+        timezoneOffset: -420,
+        tooltip: 'Los Angeles',
+      }],
+    });
+    // The hard coded date that all scheduled events should fall around
+    // Date: Sunday January 2nd, 2000 @ 00:00 PST
+    cal.setDate(new Date('2000-01-02T00:00:00'));
+  }
+
+  /**
+   * Adds a course to the calendar
+   * @param {Object} course The JSON Object for the course
+   */
+  async function addCourse(course) {
+    if (course == null) {
+      return;
+    }
+    const response = await fetch(`/api/sections?course_id=${course.course_id}`);
+    const json = await response.json();
+
+    // For now, just choose the first section out of the available ones
+    const firstSection = json.sections[0];
+    const firstMeetingInfo = firstSection.meetings[0];
+    const meetingDays = firstMeetingInfo.days;
+
+    const startTime = firstMeetingInfo.start_time;
+    const endTime = firstMeetingInfo.end_time;
+
+    if (meetingDays.includes('M')) {
+      handleCourseAddition(course, startTime, endTime, enum_days.DATE_MONDAY);
+    }
+    if (meetingDays.includes('Tu')) {
+      handleCourseAddition(course, startTime, endTime, enum_days.DATE_TUESDAY);
+    }
+    if (meetingDays.includes('W')) {
+      handleCourseAddition(
+          course, startTime, endTime, enum_days.DATE_WEDNESDAY);
+    }
+    if (meetingDays.includes('Th')) {
+      handleCourseAddition(course, startTime, endTime, enum_days.DATE_THURSDAY);
+    }
+    if (meetingDays.includes('F')) {
+      handleCourseAddition(course, startTime, endTime, enum_days.DATE_FRIDAY);
+    }
+  }
+
+  /**
+   * Adds a course to the calendar given the day of the week and a time
+   * @param {Object} course The JSON Object for the course
+   * @param {string} startTime The string representation of a 12-hour clock
+   *     time eg. 8:30pm
+   * @param {string} endTime The string representation of a 12-hour clock
+   *     time eg. 8:30pm
+   * @param {enum_days} day The day that the course falls on
+   */
+  function handleCourseAddition(course, startTime, endTime, day) {
```

Comment:
maybe rename this function to something like addCourseToSchedule -- it's not super clear what it does from the name